Source/JavaScriptCore/bytecode/Repatch.cpp

```diff
@@ -665,6 +665,38 @@ static InlineCacheAction tryCacheArrayGetByVal(JSGlobalObject* globalObject, Cod
             case ArrayStorageShape:
                 accessType = AccessCase::IndexedArrayStorageLoad;
                 break;
+            case NoIndexingShape: {
+                if (!base->isObject())
+                    return GiveUpOnCache;
+
+                if (base->structure()->mayInterceptIndexedAccesses() || base->structure()->typeInfo().interceptsGetOwnPropertySlotByIndexEvenWhenLengthIsNotZero())
+                    return GiveUpOnCache;
+
+                auto cacheStatus = prepareChainForCaching(globalObject, base, nullptr);
```

Comment:
Do we need prepareChainForCaching for indexed accesses? Some things in the loop seem like they might be needed, but others don't. For example, I don't think we need to flatten dictionaries, right? Same with looking at `propertyAccessesAreCacheable` I think. But we might need the isProxy check. I guess it's ok to be conservative and do this for now, even if it's more conservative than it needs to be.

Reply:
For now, I think this is good, getting `usesPolyProto` status etc. It is a bit conservative. I'll put FIXME for the future improvement.

Reply:
sounds good